File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
import java.util.stream.*;

class Solution {
    
    private static final int[][] persons = {
        {1, 2, 3, 4, 5},
        {2, 1, 2, 3, 2, 4, 2, 5},
        {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
    };

    public int[] solution(int[] answers) {

        int[] count = IntStream.range(0, persons.length)
            .map(i -> (int) IntStream.range(0, answers.length)
                .filter(j -> persons[i][j % persons[i].length] == answers[j])
                .count())
            .toArray();

        int max = Arrays.stream(count).max().getAsInt();

        return IntStream.range(0, count.length)
            .filter(i -> count[i] == max)
            .map(i -> i + 1)
            .toArray();
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
from collections import defaultdict


def solution(answers):
    persons = (
        (1, 2, 3, 4, 5),
        (2, 1, 2, 3, 2, 4, 2, 5),
        (3, 3, 1, 1, 2, 2, 4, 4, 5, 5)
    )
    count = defaultdict(int)
    
    for i in range(3):
        for j, answer in enumerate(answers):
            if persons[i][j % len(persons[i])] == answer:
                count[i + 1] += 1
    
    max_count = max(value for value in count.values())
    return [int(key) for key, value in count.items() if value == max_count]